### ISSUE TITLE
Issue with XMLNode.stringValue

### ISSUE THREAD

**Issue by Xpymet** [cezheng/Fuzi#48]
Hello! I'm trying to parse some html text from a website and I get strange behaviour from XMLNode.stringValue method. Part of html look like this:

```html 
...
<div class="main_text" itemprop="articleBody">
Не так давно 
<a target="_blank" href="https://somelinkhere.com">мы писали</a>
, как журналист Game Informer пытался выяснить хоть что-нибудь о судьбе 
<b>Half-Life 3</b>
, так что её просто не существует. Разговоры о третьей части в команде разработчиков действительно были, но в середине производства  
<br>
...
```

Plain text is cyrillic. When I try to print .stringValue for elements I get random \320 and \321 in text and even more. Some lines of text gets repeated multiple times. For the last part of given html print(element.stringValue) looks like that:

<pre>, так что её просто не существует. Разговоры о третьеё просто не существует. Разговоры о третье\320й части в команде разработчиков действё просто не существует. Разговоры о третье\320й части в команде разработчиков действ\320ительно были, но в середине ё просто не существует. Разговоры о третье\320й части в команде разработчиков действ\320ительно были, но в середине \320производства </pre>

Text duplicating is random each time I parse it. One time this part is OK, the other time it get all mixed up. Though .rawXML don't get those duplicates the \321\320 can be found there either. 

Using Swift, Xcode 8.2,  pod 'Fuzi', '~> 1.0.0' 

Is there any workaround? 

Thank you.

**Comment by cezheng** [comment]
Hi @Xpymet 

This might be something related to encoding. But parsing exactly the same document shouldn't result in different results on different runs. Do you request the page from the internet each time so that the server could be the problem?

**Comment by Xpymet** [comment]
Yes, I do request page each time. Page uses utf-8 and the same encoding used upon XMLDocument creation. And server is something that I have no access so no way to affect it. Html in elements after creating XMLDocument look fine, errors occur only after using .stringValue/.rawXML. Also in some part of text "&nbsp;" can be found (but not in the example I provided) though I doubt that can be the reason for errors.

**Comment by cezheng** [comment]
@Xpymet You're using XMLDocument class, right? If it is a HTML document, please use the HTMLDocument class. Or there might be some strange behaviors

**Comment by Xpymet** [comment]
Sorry, it was a typo, too late in the night when I wrote it. Of course I'm using HTMLDocument to parse page. Heres the code I use to get page and that behaviour:

```
guard let myURL = URL(string: baseUrlString + news.newsLink) else
        ...
        
        do
        {
            let mainHTMLFile    = try String(contentsOf: myURL, encoding: .utf8)
            let doc             = try HTMLDocument(string: mainHTMLFile, encoding: .utf8)
            
            for newsWrap in doc.xpath("//div[@class='lent-left']")
            {                
                print(newsWrap.stringValue) 
               ...
            }
        }
        ...
```

And it doesn't mater if I print .stringValue for whole newsWrap with a lot of text or just for one element in "articleBody". The result will be messed up.

**Comment by cezheng** [comment]
@Xpymet 
I would recommend reading the html page as Data, then use HTMLDocument(data: htmlData), in that way you don't have to specify the encoding, libxml2 will auto detect it.

**Comment by Xpymet** [comment]
Changed it as you suggested to:

```
            let mainHTMLFile    = try Data(contentsOf: myURL)
            let doc             = try HTMLDocument(data: mainHTMLFile)
            ...
            print(newsWrap.stringValue) 
```

Gave the same result with \320 and duplicate lines. Then I triple-checked everything again and found out something interesting. When I previously stated that html file is fine I checked it like that:

![print_description](https://cloud.githubusercontent.com/assets/25083739/21929079/3a9922d8-d997-11e6-8b47-0b1f8ede3909.png)

And the data looks fine! But if next you print(newsWrap) thats the moment when output goes wild with all of described errors. And all of the errors I saw before was through using print(.stringValue). After experimenting I was able to display values gotten from .stringValue inside textField without any errors. So obviously no problem with .stringValue method, just print() acting strange, will need to figure out why though. But that's a problem out of the scope of this Issue. Thank you!
